app/scripts/components/analysis/results/timeseries-data.ts

```diff
@@ -87,16 +87,26 @@ export function requestStacDatasetsTimeseries({
   end,
   aoi,
   layers,
-  queryClient
+  queryClient,
+  signal
 }: {
   start: Date;
   end: Date;
   aoi: Feature<MultiPolygon>;
   layers: DatasetLayer[];
   queryClient: QueryClient;
+  signal: AbortSignal;
 }) {
   const eventEmitter = EventEmitter();
 
+  const concurrencyManager = ConcurrencyManager();
+
+  // On abort clear the queue.
+  signal?.addEventListener('abort', () => {
```

Comment:
Shouldn't we call `removeListener` when aborting?

Reply:
Good catch! Setting up the event as a `once` thing.